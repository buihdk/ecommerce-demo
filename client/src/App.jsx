import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader/root';
import { Container } from '@material-ui/core';

import Routes from './routes';

const App = () => (
  <BrowserRouter>
    <Container fixed>
      <Routes />
    </Container>
  </BrowserRouter>
);

export default hot(App);
