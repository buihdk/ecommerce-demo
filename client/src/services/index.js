const apiServerUrl = 'http://localhost:5000';

export default {
  fetchItems: () =>
    fetch(`${apiServerUrl}/items`)
      .then(res => res.json())
      .catch(err => console.log(`Fetch problem: ${err}`)),

  fetchItemById: id =>
    fetch(`${apiServerUrl}/items/${id}`)
      .then(res => res.json())
      .catch(err => console.log(`Fetch problem: ${err}`)),

  fetchCategories: () =>
    fetch(`${apiServerUrl}/categories`)
      .then(res => res.json())
      .catch(err => console.log(`Fetch problem: ${err}`)),
};
