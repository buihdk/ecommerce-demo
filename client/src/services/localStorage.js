import { get } from 'lodash';

export const CATEGORIES = '_CATEGORIES_';

export const fetchCategories = () => {
  import('./index').then(async module => {
    if (module && module.fetchCategories) {
      const res = await module.fetchCategories();
      window.localStorage.setItem(
        CATEGORIES,
        JSON.stringify(get(res, 'data', [])),
      );
    }
  });
};

export const getCategories = () =>
  JSON.parse(window.localStorage.getItem(CATEGORIES)) || [];
