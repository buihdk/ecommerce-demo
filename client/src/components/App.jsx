import React from 'react';
import { BrowserRouter, Switch, Route, Link } from 'react-router-dom';
import { hot } from 'react-hot-loader/root';

import HomePage from '../pages/HomePage';
import ItemPage from '../pages/ItemPage';
import NotFoundPage from '../pages/NotFoundPage';
import ListItem from './ListItem';

import './App.css';

const App = () => (
  <div className="App">
    <BrowserRouter>
      <div>
        <nav>
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/item">Item</Link>
            </li>
            <li>
              <Link to="/404">Not Found</Link>
            </li>
          </ul>
        </nav>
        <Switch>
          <Route path="/item">
            <ItemPage />
          </Route>
          <Route path="/404">
            <NotFoundPage />
          </Route>
          <Route path="/">
            <HomePage />
          </Route>
        </Switch>
      </div>
    </BrowserRouter>
    <ListItem />
  </div>
);

export default hot(App);
