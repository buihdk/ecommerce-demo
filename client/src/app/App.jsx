import React, { Suspense } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { hot } from 'react-hot-loader/root';
import { Container, CircularProgress } from '@material-ui/core';

import { fetchCategories } from '../services/localStorage';
import ErrorBoundary from '../utils/ErrorBoundary';
import ErrorBar from '../utils/ErrorBar';
import Routes from '../routes';

const App = () => {
  fetchCategories();

  return (
    <BrowserRouter>
      <ErrorBoundary
        fallback={
          <ErrorBar msg="App has crashed. Check console for more info." />
        }
      >
        <Suspense fallback={<CircularProgress />}>
          <Container fixed>
            <Routes />
          </Container>
        </Suspense>
      </ErrorBoundary>
    </BrowserRouter>
  );
};

export default hot(App);
