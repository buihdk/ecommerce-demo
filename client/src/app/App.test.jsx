import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import * as LocalStorage from '../services/localStorage';
import App from './App';

const fetchCategories = jest
  .spyOn(LocalStorage, 'fetchCategories')
  .mockResolvedValue({ data: ['mock'] });

describe('App', () => {
  const wrapper = mount(<App />);

  test('renders without crashing', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(fetchCategories).toHaveBeenCalled();
  });
});
