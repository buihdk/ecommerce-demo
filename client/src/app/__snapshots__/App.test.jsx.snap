// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<App>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <WithStyles(ForwardRef(Container))
        fixed={true}
      >
        <ForwardRef(Container)
          classes={
            Object {
              "fixed": "MuiContainer-fixed",
              "maxWidthLg": "MuiContainer-maxWidthLg",
              "maxWidthMd": "MuiContainer-maxWidthMd",
              "maxWidthSm": "MuiContainer-maxWidthSm",
              "maxWidthXl": "MuiContainer-maxWidthXl",
              "maxWidthXs": "MuiContainer-maxWidthXs",
              "root": "MuiContainer-root",
            }
          }
          fixed={true}
        >
          <div
            className="MuiContainer-root MuiContainer-fixed MuiContainer-maxWidthLg"
          >
            <Routes>
              <ErrorBoundary
                fallback={
                  <UNDEFINED
                    msg="App has crashed. Check console for more info."
                  />
                }
              >
                <Suspense
                  fallback={<ForwardRef(WithStyles) />}
                >
                  <Switch>
                    <Route
                      component={[Function]}
                      computedMatch={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      exact={true}
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      path="/"
                    >
                      <component
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "createHref": [Function],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "pathname": "/",
                              "search": "",
                              "state": undefined,
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        location={
                          Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        match={
                          Object {
                            "isExact": true,
                            "params": Object {},
                            "path": "/",
                            "url": "/",
                          }
                        }
                      />
                    </Route>
                  </Switch>
                  <WithStyles(ForwardRef(CircularProgress))>
                    <ForwardRef(CircularProgress)
                      classes={
                        Object {
                          "circle": "MuiCircularProgress-circle",
                          "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
                          "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
                          "circleStatic": "MuiCircularProgress-circleStatic",
                          "colorPrimary": "MuiCircularProgress-colorPrimary",
                          "colorSecondary": "MuiCircularProgress-colorSecondary",
                          "indeterminate": "MuiCircularProgress-indeterminate",
                          "root": "MuiCircularProgress-root",
                          "static": "MuiCircularProgress-static",
                          "svg": "MuiCircularProgress-svg",
                        }
                      }
                    >
                      <div
                        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                        role="progressbar"
                        style={
                          Object {
                            "height": 40,
                            "width": 40,
                          }
                        }
                      >
                        <svg
                          className="MuiCircularProgress-svg"
                          viewBox="22 22 44 44"
                        >
                          <circle
                            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                            cx={44}
                            cy={44}
                            fill="none"
                            r={20.2}
                            strokeWidth={3.6}
                            style={Object {}}
                          />
                        </svg>
                      </div>
                    </ForwardRef(CircularProgress)>
                  </WithStyles(ForwardRef(CircularProgress))>
                </Suspense>
              </ErrorBoundary>
            </Routes>
          </div>
        </ForwardRef(Container)>
      </WithStyles(ForwardRef(Container))>
    </Router>
  </BrowserRouter>
</App>
`;
